### PR TITLE
Dashboard: fix the edit metadata form for transfers

### PR DESCRIPTION
**Pull request by replaceafill**
This PR saves the DublinCore metadata for transfers through the [DublinCoreMetadataForm](https://github.com/artefactual/archivematica/blob/e240825332b76f3cc50b66646921fd9b1ab84d4b/src/dashboard/src/components/ingest/forms.py#L23) instead of saving each field individually.

Connects to https://github.com/archivematica/Issues/issues/311
Connected to https://github.com/artefactual/archivematica/issues/1007

### CODE REVIEW

File: src/dashboard/tests/test_transfer.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        assert transfer_metadata.title == 'My First Transfer'
+        assert transfer_metadata.is_part_of == 'AIC#1234'
+        assert transfer_metadata.creator == 'The Creator'
+        assert transfer_metadata.subject == 'A Subject'
+        assert transfer_metadata.description == 'Transfer description'
+        assert transfer_metadata.publisher == 'The Publisher'
+        assert transfer_metadata.contributor == 'A contributor'
+        assert transfer_metadata.date == '2019-01-01'
+        # Verify form redirects to the metadata list after saving
+        assert response.url == 'http://testserver/transfer/{}/metadata/'.format(transfer_uuid)
```
Comment (sevein) [inline src/dashboard/tests/test_transfer.py]: Should that value use `reverse` too?

Comment (replaceafill) [reply src/dashboard/tests/test_transfer.py]: `reverse` doesn't ouput the `http://testserver` part iirc?

Comment (replaceafill) [reply src/dashboard/tests/test_transfer.py]: Maybe:

```
        redirect_url = 'http://testserver{}'.format(reverse(
            'components.transfer.views.transfer_metadata_list',
            args=[transfer_uuid],
        ))
```

File: src/dashboard/tests/test_transfer.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import os
```
Comment (ross-spencer) [inline src/dashboard/tests/test_transfer.py:2]: Something I am considering more in my testing, although I haven't had much opportunity to practice it lately, is making sure there are some Unicode strings incorporated into tests so that there is good coverage for folks wanting to use AM in different languages. You'd need an encoding declaration `# -*- coding: utf-8 -*-` first. Up to you if you'd like to add some strings in the below form to test against.

File: src/dashboard/tests/test_transfer.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        )
+        response = self.client.post(url, {
+            'title': 'My First Transfer',
+            'is_part_of': '1234',
+            'creator': 'The Creator',
+            'subject': 'A Subject',
+            'description': 'Transfer description',
+            'publisher': 'The Publisher',
+            'contributor': 'A contributor',
+            'date': '2019-01-01',
```
Comment (ross-spencer) [inline src/dashboard/tests/test_transfer.py:40]: An aside, but it is interesting the `date` field doesn't validate as a date. Nothing to fix here now, but thanks to your tests I was quickly able to try out a few different strings to see the behavior, nice :+1:

**Review by ross-spencer — APPROVED** [review]
I think this looks good @replaceafill. It's always good to see complexity reduced and new tests too. I left a comment about potentially testing some Unicode strings in the form, feel free to address that as you like. Everything passes as expected on my system. 

BTW, you should should be a member of the Artefactual organisations now? You should be good to branch from `artefactual:qa/1.x` in future. Have a chat with @sevein or maybe @scollazo if you still need access.

In terms of process, merges into `qa/1.x` will become part of the next release of Archivematica. Have a chat with @sromkey about a potential `1.9` milestone on this ticket and we will be able to merge it into `qa/1.x` prior to release so it can be tested, otherwise, we can keep this PR around for `1.10`. Once merged, we'll add the `in review` label so that it gets picked up for testing.

**Comment by ablwr** [comment]
Great work right off the bat @replaceafill!! You'll get a feel for our process during the meeting today for sure. :smile:

File: src/dashboard/tests/test_transfer.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        assert transfer_metadata.subject == 'A Subject'
+        assert transfer_metadata.description == 'Transfer description'
+        assert transfer_metadata.publisher == 'The Publisher'
+        assert transfer_metadata.contributor == 'A contributor'
+        assert transfer_metadata.date == '2019-01-01'
+        # Verify form redirects to the metadata list after saving
+        redirect_url = 'http://testserver{}'.format(reverse(
+            'components.transfer.views.transfer_metadata_list',
+            args=[transfer_uuid],
+        ))
```
Comment (sevein) [inline src/dashboard/tests/test_transfer.py:57]: Would this work?

```python
        redirect_url = response.wsgi_request.build_absolute_uri(reverse(
            'components.transfer.views.transfer_metadata_list',
            args=[transfer_uuid],
        )))
```

Comment (replaceafill) [reply src/dashboard/tests/test_transfer.py:57]: Thanks! :+1:

File: src/dashboard/tests/test_transfer.py (hunk 2 above)
Comment (replaceafill) [reply src/dashboard/tests/test_transfer.py:2]: I'll make sure to take this into account when writing tests from now on. I updated the PR to use some non-ascii characters.
